functions/src/roles/functions.ts

```diff
@@ -36,14 +36,17 @@ export const onWrite = (change: Change<DocumentSnapshot>, context: EventContext)
 
   // default to no claims
   const newUserClaims: { [id: string]: boolean } = {};
-  // go over all claims and set them
-  const afterObject = after.toObject() as { [id: string]: string[] };
-  Object.keys(afterObject).forEach(permissionGroup => {
-    (afterObject[permissionGroup]).forEach(permission => {
-      // set the data to true or null (the only two options)
-      newUserClaims[`admin.${permission}`] = true;
+
+  if (after) {
+    // go over all claims and set them
+    const afterObject = after.toObject() as { [id: string]: string[] };
```

Comment:
This type you're casting `after.toObject()` to is a little dangerous, and shouldn't be neccesary.

I guess you probably did it due to a type error when trying to do `afterObject[permissionGroup]`, in which case, whitelisting the allowed permission groups as a const would be a way to get around this issue without breaking type safety. :)